### PR TITLE
Add figure understanding sample code

### PR DESCRIPTION
**Pull request by luzhang06**

### CODE REVIEW

File: SampleCode/Python/sample_figure_understanding.ipynb [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+    "\n",
+    "![Advanced document insights with figure understanding and hierarchical document structure](../media/figure-understanding.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "- An Azure AI Document Intelligence resource in one of the 3 preview regions: **East US**, **West US2**, **West Europe** - follow [this document](https://learn.microsoft.com/azure/ai-services/document-intelligence/create-document-intelligence-resource?view=doc-intel-4.0.0) to create one if you don't have.\\n\",\n",
```
Comment (toothache) [inline SampleCode/Python/sample_figure_understanding.ipynb:19]: extra `\n`?

image

File: SampleCode/Python/sample_figure_understanding.ipynb [hunk 2]
```diff
@@ -0,0 +1,563 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Figure understanding & hierarchical document structure analysis\n",
+    "\n",
+    "This notebook demonstrates an example of using Azure AI Document Intelligence to ouptut detected figures and hierarchical document structure (in markdown). It will then crop the figures and send image content (with its caption) to Azure Open AI GPTv model to understand the semantics. The image description will be used to update the markdown output, which can be further used for [semantic chunking](https://aka.ms/doc-gen-ai).\n",
```
Comment (toothache) [inline SampleCode/Python/sample_figure_understanding.ipynb:9]: Just curious, are we calling GPT with Vision model as GPTv?

File: SampleCode/Python/sample_figure_understanding.ipynb [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+    "                    },\n",
+    "                    { \n",
+    "                        \"type\": \"image_url\",\n",
+    "                        \"image_url\": {\n",
+    "                            \"url\": data_url\n",
+    "                        }\n",
+    "                    }\n",
+    "                ] } \n",
+    "            ],\n",
+    "            max_tokens=2000 \n",
```
Comment (toothache) [inline SampleCode/Python/sample_figure_understanding.ipynb:249]: should we define the max_tokens as a CONST?

File: SampleCode/media/figure-understanding.png
Comment (toothache) [inline SampleCode/media/figure-understanding.png:1]: nit: grammar check marks `GPTv` with red squiggly line.

image

**Review by toothache — APPROVED** [review]
LGTM

File: SampleCode/Python/sample_figure_understanding.ipynb (hunk 1 above)
Comment (luzhang06) [reply SampleCode/Python/sample_figure_understanding.ipynb:19]: Resolved.

File: SampleCode/Python/sample_figure_understanding.ipynb (hunk 2 above)
Comment (luzhang06) [reply SampleCode/Python/sample_figure_understanding.ipynb]: Updated with GPT4-V

File: SampleCode/Python/sample_figure_understanding.ipynb (hunk 3 above)
Comment (luzhang06) [reply SampleCode/Python/sample_figure_understanding.ipynb:249]: added

File: SampleCode/media/figure-understanding.png
Comment (luzhang06) [reply SampleCode/media/figure-understanding.png:1]: Fixed in the new commit.

**Pull request merged**